lti_consumer/lti_1p3/consumer.py

```diff
@@ -659,3 +661,228 @@ def enable_nrps(self, context_memberships_url):
 
         # Include LTI NRPS claim inside the LTI Launch message
         self.set_extra_claim(self.nrps.get_lti_nrps_launch_claim())
+
+
+class LtiProctoringConsumer(LtiConsumer1p3):
+    """
+    This class is an LTI Proctoring Services LTI consumer implementation.
+
+    It builds on top of the LtiConsumer1p3r and adds support for the LTI Proctoring Services specification. The
+    specification can be found here: http://www.imsglobal.org/spec/proctoring/v1p0.
+
+    This consumer currently only supports the "Assessment Proctoring Messages" and the proctoring assessmen flow.
+    It does not currently support the Assessment Control Service.
+
+    The LtiProctoringConsumer requires necessary context to work properly, including data like attempt_number,
+    resource_link, etc. This information is provided to the consumer through the set_proctoring_data method, which
+    is called from the consuming context to pass in necessary data.
+    """
+    def __init__(
+        self,
+        iss,
+        lti_oidc_url,
+        lti_launch_url,
+        client_id,
+        deployment_id,
+        rsa_key,
+        rsa_key_id,
+        tool_key=None,
+        tool_keyset_url=None,
+    ):
+        """
+        Initialize the LtiProctoringConsumer by delegating to LtiConsumer1p3's __init__ method.
+        """
+        super().__init__(
+            iss,
+            lti_oidc_url,
+            lti_launch_url,
+            client_id,
+            deployment_id,
+            rsa_key,
+            rsa_key_id,
+            tool_key,
+            tool_keyset_url
+        )
+        self.proctoring_data = {}
+
+    def set_proctoring_data(self, **kwargs):
+        """
+        Set the self.proctoring_data dictionary with the provided kwargs, so long as a given key is in
+        LTI_PROCTORING_DATA_KEYS.
+        """
```

Comment:
If the kwargs are an incomplete set, what happens? Should this be more strongly typed with actual args?

Reply:
I think I did it this way originally because I thought we'd need partial data in some places, but that's no longer the case. I'll change it back to something like `set_proctoring_data(attempt_number, resource_link, ...)`.